### PR TITLE
Send ticket to unregistered ticket buyer and all ticket holders

### LINKED ISSUE #2800: Ticket buyer does not receive email with ticket
After buying a ticket as an anonymous user, the user receives an email "Welcome to eventyay.. Your account has been created".

Expected behavior would be:

**1) The buyer receives an email with the ticket attached.**

After buying a ticket as an anonmyous user, the user should receive an email:

"Your ticket to event ..."

"Hi, this is your registration confirmation for .... on ... (date)...

The event is organized by ... (organizer)...

Log in to manage your orders at http://eventyay.com.

**2) The buyer receives the registration email**

The registration email "Welcome to eventyay" as of now.

### PR DESCRIPTION
**Pull request by shubham-padia**
fixes #2800 .
Buy a ticket on http://orga3.herokuapp.com/e/64a65ca2/  with an unregistered mail, please. 

No changes to the pre-existing mail body has been made, feel free to suggest changes

selection_077

### CODE REVIEW

**Comment by SaptakS** [comment]
Please make the changes as suggested by @Marauderer97

**Comment by shubham-padia** [comment]
done @SaptakS

File: app/__init__.py [hunk 1]
```diff
@@ -410,8 +410,11 @@ def as_timezone(dt, tzname):
     """Accepts a Time aware Datetime object and a Timezone name.
         Returns Converted Timezone aware Datetime Object.
         """
-    converted_dt = dt.astimezone(timezone(tzname))
-    return converted_dt
+    if tzname is not None:
```
Comment (rafalkowalski) [inline app/__init__.py]: ``` if tzname: ``` is enough

File: app/__init__.py [hunk 2]
```diff
@@ -410,8 +410,11 @@ def as_timezone(dt, tzname):
...
+        converted_dt = dt.astimezone(timezone(tzname))
+        return converted_dt
+    else:
```
Comment (rafalkowalski) [inline app/__init__.py]: remove else

File: app/__init__.py [hunk 3]
```diff
@@ -410,8 +410,11 @@ def as_timezone(dt, tzname):
     """Accepts a Time aware Datetime object and a Timezone name.
         Returns Converted Timezone aware Datetime Object.
         """
-    converted_dt = dt.astimezone(timezone(tzname))
-    return converted_dt
+    if tzname is not None:
+        converted_dt = dt.astimezone(timezone(tzname))
+        return converted_dt
```
Comment (rafalkowalski) [inline app/__init__.py]: remove ```converted_td``` and return dt.astimezone(timezone(tzname))

Comment (shubham-padia) [inline app/__init__.py]: done @rafalkowalski

**Comment by Marauderer97** [comment]
@shubham-padia There seems to be a merge conflict on merging development branch. You may want to look into it.

**Comment by shubham-padia** [comment]
@Marauderer97 Thanks for the heads up. Didn't notice it.
@SaptakS Please review again

File: app/__init__.py [hunk 4]
```diff
@@ -415,9 +415,9 @@ def as_timezone(dt, tzname):
     """Accepts a Time aware Datetime object and a Timezone name.
         Returns Converted Timezone aware Datetime Object.
         """
-    converted_dt = dt.astimezone(timezone(tzname))
-    return converted_dt
-
+    if tzname:
+        return dt.astimezone(timezone(tzname))
+    return dt
```
Comment (shubham-padia) [inline app/__init__.py]: this was done to fix a travis build error i was getting while making the PR

File: app/helpers/ticketing.py [hunk 5]
```diff
@@ -291,26 +291,6 @@ def initiate_order_payment(form, paid_via=None):
...
-                holder_user = TicketingManager.get_or_create_user_by_email(holders_emails[i], data)
-                ticket_holder = TicketHolder(firstname=data['firstname'],
-                                             lastname=data['lastname'],
-                                             ticket_id=int(holders_ticket_ids[i]),
-                                             email=holder_user.email,
-                                             order_id=order.id)
-                DataManager.add_attendee_role_to_event(holder_user, order.event_id)
-                db.session.add(ticket_holder)
-
             user = TicketingManager.get_or_create_user_by_email(email, form)
```
Comment (shubham-padia) [inline app/helpers/ticketing.py]: moved down because other order details were needed while sending the mail

Comment (SaptakS) [inline app/helpers/ticketing.py]: @shubham-padia, this was moved up to ensure that even if payment is not completed but the Pay Now button is clicked, then attendees are added to the DB. So the attendees are added to the DB with a initiated status even when payment isn't completed. Can you please ensure that this feature still works? Once you ensure this, the remaining seems good to me.

Comment (shubham-padia) [inline app/helpers/ticketing.py]: > ensure that even if payment is not completed but the Pay Now button is clicked, then attendees are added to the DB

@SaptakS Yep. It works :smile:

**Pull request merged**